src/app/modules/form/admin-form/__tests__/admin-form.service.spec.ts

```diff
@@ -2242,4 +2246,106 @@ describe('admin-form.service', () => {
       expect(expected._unsafeUnwrapErr()).toBeInstanceOf(DatabaseError)
     })
   })
+
+  describe('shouldUpdateFormField', () => {
+    const MOCK_FORM = {
+      admin: {
+        _id: new ObjectId(),
+      },
+    } as unknown as IPopulatedForm
+
+    const MOCK_UNVERIFIABLE_MOBILE_FIELD = generateDefaultField(
+      BasicField.Mobile,
+    )
+    const MOCK_VERIFIABLE_MOBILE_FIELD = generateDefaultField(
+      BasicField.Mobile,
+      { isVerifiable: true },
+    )
+
+    const countSpy = jest.spyOn(SmsService, 'retrieveFreeSmsCounts')
+
+    it('should return the form when admin is under the limit', async () => {
+      // Arrange
+      countSpy.mockReturnValueOnce(okAsync(SMS_VERIFICATION_LIMIT))
+
+      // Act
+      const actual = await shouldUpdateFormField(
+        MOCK_FORM,
+        MOCK_VERIFIABLE_MOBILE_FIELD,
+      )
+
+      // Assert
+      expect(actual._unsafeUnwrap()).toEqual(MOCK_FORM)
+    })
+
+    it('should return the form when the form has a message service name', async () => {
+      // Arrange
+      const MOCK_ONBOARDED_FORM = { ...MOCK_FORM, msgSrvcName: 'form a form' }
+
+      // Act
+      const actual = await shouldUpdateFormField(
+        MOCK_ONBOARDED_FORM,
+        MOCK_VERIFIABLE_MOBILE_FIELD,
+      )
+
+      // Assert
+      expect(countSpy).not.toHaveBeenCalled()
+      expect(actual._unsafeUnwrap()).toEqual(MOCK_ONBOARDED_FORM)
+    })
+
+    it('should return the form when mobile field is not verifiable', async () => {
+      // Act
+      const actual = await shouldUpdateFormField(
+        MOCK_FORM,
+        MOCK_UNVERIFIABLE_MOBILE_FIELD,
+      )
+
+      // Assert
+      expect(countSpy).not.toHaveBeenCalled()
+      expect(actual._unsafeUnwrap()).toEqual(MOCK_FORM)
+    })
+
+    it('should return the form when the field provided is not a mobile field', async () => {
+      // Arrange
+      const MOCK_NUMBER_FIELD = generateDefaultField(BasicField.Number)
+
+      // Act
+      const actual = await shouldUpdateFormField(MOCK_FORM, MOCK_NUMBER_FIELD)
+
+      // Assert
+      // The sms counts check is only executed when it is a mobile field
+      expect(countSpy).not.toHaveBeenCalled()
+      expect(actual._unsafeUnwrap()).toEqual(MOCK_FORM)
+    })
+
+    it('should return sms retrieval error when sms limit exceeded and admin is attempting to toggle sms verification for mobile field on', async () => {
```

Comment:
this test description should not be "coupled" to the actions of the client, i think the first part is enough